src/PokemonTrainer.java

```diff
@@ -0,0 +1,11 @@
+public class PokemonTrainer extends PokemonGymImpl {
```

Comment:
Je hebt hier extends PokemonGymImpl, maar je implementeert de methodes uit deze interface niet. Een pokemontrainer is ook een persoon en geen gym.  Je had hier je pokemons kunnen implementeren.